models/note.go

```diff
@@ -22,16 +23,33 @@ var categoryStrings = [...]string{
 	"predictions",
 }
 
-func (category Category) String() string {
-	if category < MARGINALIA || category > PREDICTIONS {
-		return "Unknown"
+var UnDeserializeableCategoryStringError = errors.New("String does not correspond to a Note Category")
+
+func DeserializeCategory(input string) (Category, error) {
+	for i := 0; i < len(categoryStrings); i++ {
```

Comment:
Is it weird that we're doing a linear search here? Maybe I was wrong about the front-end not knowing about numbers we have corresponding to each category? Because something about this feels off

Reply:
is it weird that we are searching? or is it weird that the search is linear? Because if you want the front-end to be storage agnostic (which makes sense to me). Then you'll end up having to do a search. Though if you want we can store them in fancy objects to make it smaller than linear search but that seems like an over optimization at this point.

Reply:
i'm not worried about the performance. it just feels like we're doing something wrong. this is just making me think that the front-end should maybe get like a map of ids to categories and then we never have to worry about converting from string -> category or vice versa anymore

Reply:
That's fair. Let's postpone this conversation until we are looking at the pull request for the front end. Since for the backend this is a pretty simple fix.